src/HttpServer.php

```diff
@@ -255,4 +251,10 @@ function () use ($connection) {
             }
         );
     }
+
+    /** @internal */
+    public function sendResponse(ResponseInterface $response, ConnectionInterface $connection)
```

Comment:
Is this related? Should this be a separate PR? Also looks like some tests were missing here?